specs/CommandBar/new-command-bar-features.md

```diff
@@ -0,0 +1,349 @@
+
+```xaml
+<AppBarButton>View
+    <AppBarButton.Flyout>
+        <MenuFlyout>
+            <MenuFlyoutItem>Open</MenuFlyoutItem>
+            <MenuFlyoutSeparator/>
+            <RadioMenuFlyoutItem>Landscape</RadioMenuFlyoutItem>
+            <RadioMenuFlyoutItem IsChecked="True">Portrait</RadioMenuFlyoutItem>
+            <MenuFlyoutSeparator/>
+            <RadioMenuFlyoutItem>Small icons</RadioMenuFlyoutItem>
+            <RadioMenuFlyoutItem IsChecked="True">Medium icons</RadioMenuFlyoutItem>
+            <RadioMenuFlyoutItem>Large icons</RadioMenuFlyoutItem>
+        </MenuFlyout>
+    </AppBarButton.Flyout>
+</AppBarButton>
+```
+
+![Image of a menu flyout with radio items](images/radiomenuflyoutitems.png)
+
+The issue arises when you want to combine these two concepts. It would be useful to have `RadioMenuFlyoutItems` have 
+their own child items that participate in the same Group as their parent.
+This would allow for cleaner UIs by decreasing the size of `MenuFlyouts` that have a long list of `RadioMenuFlyoutItems`. 
+See an example below:
+
+![Nested radio menu flyout example](images/commandbar2.png)
+
+### Issue #3: CommandBarFlyout does not have an "always expanded" state.
+
+Currently, a `CommandBarFlyout` with secondary commands always shows a […] "more" button that
+allows the user to collapse the `CommandBarFlyout` into just the top bar (primary commands). 
+There's currently no way to have a `CommandBarFlyout` with secondary commands that is always expanded when invoked. 
+
+![Command bar with more button](images/more-button.jpg)
+
+When the `CommandBarFlyout` is invoked via touch, only the primary commands are shown (with the option for the user to
+expand the list of secondary commands). The developer is currently not able to control how the `CommandBarFlyout` appears when invoked, 
+and cannot control whether it can be collapsed or not. 
+
+# Description
+
+### Feature #1. Built-in styles to support adding `SplitButton` to `CommandBar` and `CommandBarFlyout`
+
+WinUI will provide built in styles for SplitButton that will allow them to conform with the look and feel of an `AppBarButton`. 
+These styles will change properties such as the `SplitButton`'s height/width, font size, color, interactions (color changes), and more. 
+
+There will be two styles provided: `CommandBarSplitButtonStyle` and `CommandBarFlyoutSplitButtonStyle`. 
+These styles can be accessed as ThemeResources in any WinUI app. 
+
+Note: if a SplitButton item inside of a CommandBar enters the CommandBar's overflow menu, it will pick up the `CommandBarFlyoutSplitButtonStyle`.
+
+For example:
+
+```xml
+<CommandBarFlyout x:Name="ImageCommandsFlyout" >  
+    <AppBarButton Icon="Copy" ToolTipService.ToolTip="Copy" Label="Copy"/>
+
+    <CommandBarFlyout.SecondaryCommands>
+        <AppBarElementContainer>
+            <SplitButton ToolTipService.ToolTip="Insert"
+                         Style="{ThemeResource SplitButtonCommandBarFlyoutStyle}">
```

Comment:
Is there some way to get the styling automatically, or does everybody have to remember to opt into the "correct" style manually? Would it be overkill to have a `AppBarSplitButton` element that applies the style for you?

```csharp
unsealed runtimeclass AppBarSplitButton : SplitButton
{
  // no new members. Default Style is SplitButtonCommandBarFlyoutStyle.
}
```

Reply:
We could have a setter inside the CommandBarFlyout resources for the SplitButton (and other controls). Those styles would only apply to the controls inside the CommandBarFlyout (I hope, flyouts are always a bit tricky in that area). That way we wouldn't need wrappers of have people remember to opt into the correct style.

Reply:
Good callout - when you place a SplitButton in the CommandBar or CommandBarFlyout, the correct style will automatically be applied.

Reply:
Discussion: Yes, when you put these controls in a CommandBar, it automatically picks up the style. So the only thing special you need to know to use these controls is to put them into the AppBarElementContainer.

Recommend: Update to explain that, and take it out of the example code.